### DISCUSSION TITLE
generate proper characters of ion channel embeded in a membrane

### DISCUSSION THREAD

**Discussion by linda091021** [Electrostatics/apbs#169] [Q&A]
Dear all,
How could I get the proper electric charge and electrostatic potential of an ion channel through PDB2PQR and APBS. 
I did't notice any options to set a membrane environment on this server https://server.poissonboltzmann.org/.

or the electric charge is not influenced by the environment? The .pqr file generated is all right?

In that case, how could I generate a membrane environment to calculate electrostatic potential of an ion channel, which should be definitely influenced by solvent or membrane. I did't find any options in the server or plugin in PyMOL.

At this moment, I generate the electrostatic potential through plugin in PyMOL. 
As there is a pore in the center of the ion channel, if I only focus on the electrostatic potential of this pore region, could I think the results around the pore region is reliable?
 I'm wondering  if the pore region is considered full of solvent with dielectric constant=2 in the APBS calculation, even if I hadn't defined a cylinder region to represent the pore region?

Thanks!

**Comment by sobolevnrm** [comment]
Hello --

I'm sorry but we do not currently have automatic options for setting up membrane environments around a protein as part of the main APBS package.  However, Michael Grabe has developed the [APBSmem package](https://apbsmem.sourceforge.io/) which may do something very similar to what you're asking.

Thank you,

Nathan

**Reply by linda091021** [reply]
Hi, Nathan

Thank you so much for the kind suggestions and quick reply!

I'll try the the APBSmem package you mentioned.

Best Wishes!

Linlin

**Reply by willemsk** [reply]
@sobolevnrm I was wondering if you would consider a pull request on integrating the membrane functionality of the [`draw_membrane2a`](https://github.com/Becksteinlab/BornProfiler/blob/develop/src/drawmembrane/draw_membrane2a.c) program into the main APBS code.
As you know, the current workflow requires dumping a lot of data to disk, modifying it, and loading it back into memory. This is often more time-consuming than the actual focusing step.

If it is something you would be open to, I am willing to work on this and draft a first pull request

**Reply by sobolevnrm** [reply]
Hi @willemsk -- can you open a PR with an outline of what you're thinking rather than an entire implementation?  I'm not sure how clean the integration with the main code would be, so I have reservations.  I don't want you to waste effort, if this is unlikely to be merged.  Thank you, Nathan.
